File: server/model/workspace.go
package model

type Workspace struct {
	ID          string                 `json:"id"`
	SignupToken string                 `json:"signupToken"`
	Settings    map[string]interface{} `json:"settings"`
	ModifiedBy  string                 `json:"modifiedBy"`
	UpdateAt    int64                  `json:"updateAt"`
}

File: webapp/src/blocks/workspace.ts
// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
interface IWorkspace {
    readonly id: string,
    readonly title: string,
    readonly signupToken: string,
    readonly settings: Readonly<Record<string, any>>
    readonly modifiedBy?: string,
    readonly updateAt?: number,
}

export {IWorkspace}
